beefy-gadget/src/worker.rs

```diff
@@ -304,7 +305,13 @@ where
 			return false;
 		}
 
-		let diff = self.best_finalized_block.saturating_sub(self.best_block_voted_on);
+		let best_finalized = if let Some(block) = self.best_finalized_block {
+			block
+		} else {
+			return false;
```

Comment:
I'd add `debug/trace` logs in every `return false` so that we can easily figure out why some validator is not voting.

Reply:
Done